src/commands/site-list.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace Team51\Command;
+
+use Team51\Helper\API_Helper;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Helper\ProgressBar;
+
+class Site_List extends Command {
+	protected static $defaultName = 'site-list';
+
+	protected function configure() {
+		$this
+		->setDescription( 'Shows list of public facing sites managed by Team 51.' )
+		->setHelp( 'Use this command to show a list of sites and summary counts managed by Team 51.' );
+	}
+
+	protected function execute( InputInterface $input, OutputInterface $output ) {
+		$api_helper = new API_Helper;
+
+		$output->writeln( '<info>Fetching sites...<info>' );
+
+		$all_sites = $api_helper->call_wpcom_api( 'rest/v1.1/me/sites?fields=ID,name,URL,is_private,is_coming_soon,is_wpcom_atomic,jetpack', array() );
+
+		if ( empty( $all_sites ) ) {
+			$output->writeln( '<error>Failed to fetch sites.<error>' );
+			exit;
+		}
+
+		$site_count = count( $all_sites->sites );
+		$output->writeln( "<info>{$site_count} sites found in total. Filtering...<info>" );
+
+		$all_sites = $all_sites->sites;
+
+		$ignore = array(
+			'staging',
+			'jurassic',
+			'wpengine',
+			'wordpress',
+			'develop',
+			'com/',
+			'org/',
+			'mdrovdahl',
+			'/dev.',
+		);
+
+		$free_pass = array(
+			'wpspecialprojects.wordpress.com',
+			'tumblr.wordpress.com',
+			'tonyconrad.wordpress.com',
+		);
+
+		$filtered_sites = array();
+		foreach ( $all_sites as $site ) {
+			$found = false;
+			foreach ( $ignore as $word ) {
+				if ( false !== strpos( $site->URL, $word ) ) {
+					$passed = false;
+					foreach ( $free_pass as $pass ) {
+						if ( false !== strpos( $site->URL, $pass ) ) {
+							$passed = true;
+							break;
+						}
+					}
+					if ( false === $passed ) {
+						$found = true;
+						break;
+					}
+				}
+			}
+			if ( false === $found && false === $site->is_private ) {
+				$filtered_sites[] = $site;
+			}
+		}
+
+		$filtered_site_list = array_filter(
+			$filtered_sites,
+			function( $site ) {
+				return false === $site->is_coming_soon;
+			}
+		);
+
+		$final_site_list = array();
+		foreach ( $filtered_site_list as $site ) {
+			if ( true === $site->is_wpcom_atomic ) {
+				$server = 'Atomic';
+			} elseif ( true === $site->jetpack ) {
```

Comment:
@mdrovdahl Yeah, this is a tricky one since we have no way of knowing who the actual host is. The summary at the end of the list does show it as `Pressable (or other) sites` to highlight the fact. Any thoughts on a better way to show it in the table are welcome. Perhaps just `Pressable (or other)`?

Btw, I just pushed a commit that creates a csv file as `sites.csv` in the team51 root folder. It's still a work in progress though (I'm jumping over to help with the bloated dev requests queue for the moment), and will currently create the full table in csv form when the command is run. When I get back to it I will work on the options you mentioned (optional csv/json, excluding columns, etc).

Reply:
@mdrovdahl Thinking about this a bit more... I could pull a list of URLs from Pressable and compare it to my "Pressable (or other)" list, and any not matching would (or should) be truly "Other hosting". I was going to ask if this was a good feature to have, but typing it out kind of answers the question, and sounds like a good metric to have. I'm going to add it to the to-do list unless you really think it's not worth it.